arches_controlled_lists/src/arches_controlled_lists/components/editor/ItemEditor.vue

```diff
@@ -17,20 +17,56 @@ import {
 import type { Ref } from "vue";
 import type { ControlledListItem } from "@/arches_controlled_lists/types";
 
+const prefLabelEditor = useTemplateRef("prefLabelEditor");
+const altLabelEditor = useTemplateRef("altLabelEditor");
+const noteEditor = useTemplateRef("noteEditor");
+const uriEditor = useTemplateRef("uriEditor");
+const imageEditor = useTemplateRef("imageEditor");
+
 const { displayedRow: item } = inject(displayedRowKey) as unknown as {
     displayedRow: Ref<ControlledListItem>;
 };
 provide(itemKey, item);
+
+const isEditing = inject("isEditing") as Ref<boolean>;
+
+watch(
+    [
+        () => prefLabelEditor.value?.isEditing,
+        () => altLabelEditor.value?.isEditing,
+        () => noteEditor.value?.isEditing,
+        () => uriEditor.value?.isEditing,
+        () => imageEditor.value?.isEditing,
+    ],
+    () => {
+        isEditing.value =
+            prefLabelEditor.value?.isEditing ||
+            altLabelEditor.value?.isEditing ||
+            noteEditor.value?.isEditing ||
+            uriEditor.value?.isEditing ||
+            imageEditor.value?.isEditing ||
+            false;
+    },
+);
```

Comment:
why do this at this level? `isEditing` is already being provided, why not inject it into the editor components instead and lose the `defineExpose`?

Reply:
Yeah this was tricky: there are multiple editors on the same page. In fact you can have infinite image metadata editors. I didn't want to be in a situation where one of the image metadata editors was querying the other n image metadata editors to find out if it was appopriate to flip isEditing to False.

I thought of generating ids for each editor and updating some global state, but I found this more self-documenting and cleaner.

Open to other approaches for sure!

Reply:
After discussion we decided to prevent initiating multiple edits.